src/test/unit/algorithms/SegmentTest.cpp

```diff
@@ -0,0 +1,172 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+ * Implementation of unit tests for Segment
+ */
+
+#include <nupic/algorithms/Segment.hpp>
+#include <gtest/gtest.h>
+#include <set>
+
+using namespace nupic::algorithms::Cells4;
+using namespace std;
+
+
+void setUpSegment(Segment &segment, 
+                  vector<UInt> &inactiveSegmentIndices,
+                  vector<UInt> &activeSegmentIndices,
+                  vector<UInt> &activeSynapseIndices,
+                  vector<UInt> &inactiveSynapseIndices)
+{
+  vector<double> permanences = {0.2, 0.9, 0.9, 0.7, 0.4, // active synapses
+                                0.8, 0.1, 0.2, 0.3, 0.2}; // inactive synapses
+
+  set<UInt> srcCells;
+  for (uint i = 0; i < permanences.size(); i++)
```

Comment:
@SaganBolliger uint is not defined (as seen here https://ci.appveyor.com/project/numenta-ci/nupic-core/build/0.3.0.555#L459). Suggestions are uint from capnp and kj, or just replace this with `UInt` from nupic namespace.